pkg/limits/consumer_test.go

```diff
@@ -209,3 +210,126 @@ func TestConsumer_ReadinessCheck(t *testing.T) {
 	u.Iter(func(_ string, _ int32, _ streamUsage) { n++ })
 	require.Equal(t, 2, n)
 }
+
```

Comment:
not sure if these tests are necessary and/or redundant with existing tests, AI had me add them and I need to look a little closer still